internal/cmd/shell.go

```diff
@@ -80,17 +80,54 @@ coder sh front-end-dev cat ~/config.json`,
 	}
 }
 
+// shellEscape escapes an argument so that we can pass it 'sh -c'
+// and have it do the right thing.
+//
+// Use this to ensure that the result of a command running in
+// the development environment behaves the same as the command
+// running via "coder sh".
+//
+// For example:
+//
+//   $ coder sh env
+//   $ go run ~/test.go 1 2 "3 4" '"abc def" \\abc' 5 6 "7 8 9"
+//
+// should produce the same output as:
+//
+//   $ coder sh go run ~/test.go 1 2 "3 4" '"abc def" \\abc' 5 6 "7 8 9"
+func shellEscape(arg string) string {
+	r := strings.NewReplacer(`\`, `\\`, `"`, `\"`, `'`, `\'`, ` `, `\ `)
+	return r.Replace(arg)
```

Comment:
I think eventually we should be relying on shlex (or similar) to be parsing and escaping shell commands from strings into arg arrays, and avoid rolling our own for things like this.

Reply:
@tychoish Agree, I did a quick search bit didn't find any good ones. Happy to switch over now, or we can do it later, WDYT?

Reply:
https://github.com/google/shlex is functional. 

I think this might also solve the %q question in the other PR

Reply:
@tychoish Oh yeah, that looks much better. Looks like there were a few cases that I missed.

I don't think this will work for #226 because we need the shell to interpret the subshell calls for getent/etc, but I will give it a try

Reply:
Hmm it looks like shlex is for parsing shell expressions, how do I use it to escape things?